static/app/views/settings/account/notifications/notificationSettingsByType.tsx

```diff
@@ -0,0 +1,357 @@
+        notificationSettings,
+        updatedNotificationSettings
+      ),
+    });
+
+    return updatedNotificationSettings;
+  };
+
+  getStateToPutForParent = (changedData: {[key: string]: string}, parentId: string) => {
+    const {notificationType} = this.props;
+    const {notificationSettings} = this.state;
+
+    const updatedNotificationSettings = getStateToPutForParent(
+      notificationType,
+      notificationSettings,
+      changedData,
+      parentId
+    );
+
+    this.setState({
+      notificationSettings: mergeNotificationSettings(
+        notificationSettings,
+        updatedNotificationSettings
+      ),
+    });
+    return updatedNotificationSettings;
+  };
+
+  /* Methods responsible for rendering the page. */
+
+  getGroupedParents = (): {[key: string]: Organization[] | Project[]} => {
+    /**
+     * The UI expects projects to be grouped by organization but can also use
+     * this function to make a single group with all organizations.
+     */
+    const {organizations} = this.props;
+    const {projects: stateProjects} = this.state;
+
+    return this.isGroupedByProject()
+      ? Object.fromEntries(
+          Object.values(
+            groupByOrganization(stateProjects)
+          ).map(({organization, projects}) => [`${organization.name} Projects`, projects])
+        )
+      : {organizations};
+  };
+
+  getParentField = (parent: Organization | Project): FieldObject => {
+    const {notificationType} = this.props;
+
+    const defaultFields = NOTIFICATION_SETTING_FIELDS[notificationType];
+    const currentDefault = Object.values(this.getUserDefaultValues())[0];
+
+    return Object.assign({}, defaultFields, {
+      label: (
+        <FieldLabel>
+          <Avatar
+            {...{[this.isGroupedByProject() ? 'project' : 'organization']: parent}}
+          />
+          <span>{parent.slug}</span>
+        </FieldLabel>
+      ),
+      getData: data => this.getStateToPutForParent(data, parent.id),
+      name: parent.id,
+      choices: defaultFields.choices?.concat([
+        [
+          'default',
+          `${t('Default')} (${getChoiceString(defaultFields.choices, currentDefault)})`,
+        ],
+      ]),
+      defaultValue: 'default',
+      help: undefined,
+    }) as any;
+  };
+
+  getInitialData(): {[key: string]: string} {
+    const {notificationType} = this.props;
+
+    const providerList = this.getCurrentProviders();
+    const initialData = {
+      [notificationType]: providerList.length
+        ? this.getUserDefaultValues()[providerList[0]]
+        : 'never',
+    };
+
+    if (!this.isEverythingDisabled()) {
+      initialData.provider = providerListToString(providerList);
+    }
+    return initialData;
+  }
+
+  getFields(): FieldObject[] {
+    const {notificationType} = this.props;
+
+    const fields = [
+      Object.assign(NOTIFICATION_SETTING_FIELDS[notificationType], {
+        help: t('This is the default for all projects.'),
```

Comment:
I'm seeing this as the text below "Issue alert notifications" always

Reply:
This looks like `Object.assign` overwriting the constant. :C